### ISSUE TITLE
Instant Server Crash (Loop)

### ISSUE THREAD

**Issue by KontraM8** [ZeroNoRyouki/ExtremeReactors2-issues#10]
Sooo I connected my Reinforced 5x5x5 Reactor with my Mekanism Turbine via Advanced Pressurized Tube and I got an instant Crash. I cant start my server now, because they are Connected. The tube is set on Pull on the Reinforced Active Fluid Port.

**Comment by KontraM8** [comment]
This is the Crash Report that Iam getting: https://pastebin.com/nde5HfLe

**Comment by ZeroNoRyouki** [comment]
It's fixed in 2.0.8. Please update ER and let me know

**Comment by KontraM8** [comment]
It kinda works. Now I got a new problem: When breaking the Access Port, I dont get it back. On top of that, it doesnt connect at all

**Comment by KontraM8** [comment]
I tried it with a Mechanical Pipe, I indeed get the Steam out of it, but now the Tube doesnt Connect to the Turbine, as it is a Fluid and not a Gas

**Comment by ZeroNoRyouki** [comment]
> It kinda works. Now I got a new problem: When breaking the Access Port, I dont get it back. On top of that, it doesnt connect at all

Are you talking about the Solid Access Port ? To what are you trying to connect it?

**Comment by KontraM8** [comment]
Reactor Active Forge Fluid Port (Legacy)

**Comment by ZeroNoRyouki** [comment]
> I tried it with a Mechanical Pipe, I indeed get the Steam out of it, but now the Tube doesnt Connect to the Turbine, as it is a Fluid and not a Gas

What kind of Fluid Port are you using in the Turbine? Active or Passive?

**Comment by KontraM8** [comment]
Neither, Iam using the Mekanism Turbine as mentioned before

**Comment by papaworld** [comment]
Mekanism Turbine uses GAS Steam not Liquid Steam... (I got the same crash). You can convert ER2 Liquid Steam to Meka Gas Steam but that would take lots of Rotary Condenserators with full upgrades... which is not cost effective to achieve a decent Gas Steam amount to turn the Turbine from Mekanism.

**Comment by KontraM8** [comment]
Fun Fact: The Fluid doesnt appear to get into the Rotary Condensentrator

**Comment by ZeroNoRyouki** [comment]
> Neither, Iam using the Mekanism Turbine as mentioned before

Ah that's it then. The Forge Fluid Port use Forge own IFluidHandler system. I'm working on a separate Fluid Port for Mekanism to handle it's Gas interface but I'm still debading on the conversion system from the vapors produces by the Reactor to Mek gasses

**Comment by papaworld** [comment]
I got the ER2 steam to convert in a rotary condenserator at 50mb with full upgrades. Totally not feasible.

**Comment by ZeroNoRyouki** [comment]
The fact that Mekanism provide it's own way to convert from liquid (steam) to gas (steam) it's one of the thing that pushed me to put the Mek Fluid Port on hold. Not sure if they would get pissed if I started converting (liquid) steam on the fly

**Comment by KontraM8** [comment]
My turbine only has an Input of 400mb/t meanwhile iam producing way more than that? wth am i doing wrong

**Comment by ZeroNoRyouki** [comment]
> It kinda works. Now I got a new problem: When breaking the Access Port, I dont get it back. On top of that, it doesnt connect at all

As for the port don't dropping, your are right... I so hate MC new block drop system, it's so damn stupid to have to add a json file to make a block drop itselft uff.

**Comment by papaworld** [comment]
There are 4 systems I am aware of to produce Mekanism gas steam... rotary condenserator, fusion reactor, fission reactor and boiler. As I was looking forward to your 1.16.x updates for the release of the turbine... I was hoping you would maybe changet your liquid steam system to gas. Maybe instead of "converting" liquid steam to gas... the reactor would just generate gas steam as an output.

**Comment by KontraM8** [comment]
Thats what I was thinking and considering doing, but well... guess that doesnt work as I hoped

**Comment by KontraM8** [comment]
And on thing I noticed, at least for me, my Reactor Fuel is never at 100%

**Comment by ZeroNoRyouki** [comment]
> There are 4 systems I am aware of to produce Mekanism gas steam... rotary condenserator, fusion reactor, fission reactor and boiler. As I was looking forward to your 1.16.x updates for the release of the turbine... I was hoping you would maybe changet your liquid steam system to gas. Maybe instead of "converting" liquid steam to gas... the reactor would just generate gas steam as an output.

The thing is, the Reactor does not generate steam. It convert a registered coolant into a given amount of it's registered vapor (and the Turbine condensate the vapor into it's source coolant)

Coolant and Vapors, and their transformations, are added using the API of ER. ER add to itself water as a coolant that vaporize to steam (and viceversa) but other mods can add their fluid too.

So an intermediate layer is needed to convert a vapor to a Mek gas like I've said above

**Comment by ZeroNoRyouki** [comment]
> And on thing I noticed, at least for me, my Reactor Fuel is never at 100%

Yup, the SOLID Access Port is bad at math :)

**Comment by KontraM8** [comment]
Sooo Iam guessing theres no way to change that? :D

**Comment by papaworld** [comment]
Thank you Zero for clarity and BIG thank you for porting ER to ER2.

Kontra... what size turbine is locked in to 400mb/t... sounds like it is a basic smaller multiblock which would limit the input/output. Build up you turbine with reinforced to achieve higher steam input for higher FE output.

The Reactor fuel rods must be full to achieve highest output, check for fuel ingots in the solid access port, you should have atleast 1 ingot. They will never be 100% on a running system because of waste when waste amount increases the reactor output decreases. The smaller the reactor the more amplified the decreased output.

**Comment by ZeroNoRyouki** [comment]
> Sooo Iam guessing theres no way to change that? :D

Who knowns, maybe in the future there would be a better way to convert ingots to reactants :)

**Comment by ZeroNoRyouki** [comment]
I'll ask the Mekanisms guys what they would think of me converting liquid Steam to Gas Steam, and thoretically, any fluid to gas

**Comment by ZeroNoRyouki** [comment]
> Thank you Zero for clarity and BIG thank you for porting ER to ER2.

You are welcome!

**Comment by KontraM8** [comment]
@papaworld Iam using the Turbine from Mekanism, it is indeed a Basic Small Multiblock 
![Bild_2020-11-24_190429](https://user-images.githubusercontent.com/74981698/100134035-e0af9c80-2e87-11eb-847a-1787434bd99f.png)

**Comment by KontraM8** [comment]
But thanks for your help so far Guys!

**Comment by papaworld** [comment]
Kontra, if I may suggest for your consideration.... because your Mekanism Turbine will not work with ER2 maybe build an ER2 full size Turbine. Go to the config file for ER2. Change Power Produced from 1.0 to something like 10.0 and also for fairness change the fuel use also from 1.0 to 10.0 to maintain balance. In 1.12.2 the turbine maxed out at multiplier 35.0 for 1MRF/t output using 1 RF port and the Turbine generated over 1MRF at the coils.

**Comment by KontraM8** [comment]
I think Ill just use the ER2 Reactor for RF/t to boost my future Boiler which then provides the Steam to my Turbine. I indeed set the config higher already :D

**Comment by papaworld** [comment]
Good luck with that. In 1.12.2 I had an ER 11x11x9 50,000mb/t steam reactor directly feeding the Mekanism Turbine 5x5x8 producing 573,000RF/t which returned water back to the reactor.

Hey when you get it to work... would you mind posting it?

**Comment by KontraM8** [comment]
If you want, we can stay In contact on Discord : KontraM8:7156 , this is not the Chat for it:)

**Comment by papaworld** [comment]
Gotzit! Thank you Sir.

**Comment by ZeroNoRyouki** [comment]
I've asked there if you would like to see how that would turn out : https://github.com/mekanism/Mekanism/issues/6741

**Comment by papaworld** [comment]
Hey Ya Zero... saw the posts on Meka6741. BTW,  pupnewfster is really a nice guy. The problem with the rotary condenserator is it will convert liquid steam to gas steam, but, there is a "not worth it" situation.

One Rotary Condenserator will need 8 speed and 8 efficiency upgrades to convert liquid steam to gas steam at a maximum rate of 256mb/t (way too little for ER2 turbines) and requires 51,200FE/t to do it... yeah 51KFE.

BTW, ER2 liquid steam does work in Meka Rotary Cond to produce gas steam.

I do have some thoughts tho, if you would wanna consider them.

**Comment by ZeroNoRyouki** [comment]
> BTW, ER2 liquid steam does work in Meka Rotary Cond to produce gas steam.

Yup, It's tagged as "steam" so it should works in any mods that accecpt a fluid tagged as such

> I do have some thoughts tho, if you would wanna consider them.

Fire away!

**Comment by papaworld** [comment]
Looking back to 1.7.10 - how mods boiled water
- MineFactory Reloaded Boiler: 1 part water > 4 parts steam - 4 buckets
- IndustrialCraft2 Boiling System: 1 part water > ~6 parts steam (not superheated) - 6 buckets
- RailCraft: 1 part water > ~ 7 parts steam - 7 buckets
- Big Reactors and Mekanism: 1 part water > 1 part steam - 1 bucket
- - Real Life: 1 part water > 17 parts steam (normal temp [212F] normal pressure [0 Bar - no backpressure])

Mekanism 1.7.10 wiki states (paraphrasing from memory): The 5x5x8 Industrial Turbine will output the same amount of RF/t as the Big Reactors Turbine with the same amount of steam injected. (I''m not trying to discredit or step on anyones feet)

Personally, I believe that BR Turbines were a novel idea, but fell short for the RF output vs size and cost. This also carried over in ER for 1.12.2. So I did one of three things for world design. Raised the RF production output multiplier and matched that number to Reactor fuel use multiplier so they matched. Got 1MRF output at 7x7x16 Turbine RF Port and just over 1MRF at the coils. Perfect! Challenging creativity for Fuel Ingots - got it! The second thing, match an ER Reactor 11x11x9 (50,000mb steam) to Meka Turbine 5x5x8, it works very well, producing 573,000RF/t (2 coils). The third, scraped the ER and Meka systems and went with something else that produced over 24KRF in a 5x5x2 area.

BTW, the Meka 5x5x8 Turbine is NOT limited to 573,000RF/t... that is only at 2,000mb/t steam in. This small Turbine will max out at 3.2MRF/t with enough steam in.

With that all said, it begs the question... Why would someone want to connect a Meka Turbine to EF Reactor?

Only one answer: Power Output! using ER and Meka stock confings. 1 small 5x5x8 Turbine.

My thoughts are to re-think how ER2 Turbines work, not their design, persay. Steam rate input vs RPM vs qty coils vs RF (FE) output vs water return.

Maybe an update on Turbine math equations will bring ER2 Turbines up so the whole idea of Meka Turbines is not necessary at all and mute.

This comment is information background for concept consideration, i would be happy to and enjoy assisting in application.

**Comment by ZeroNoRyouki** [comment]
Well, I think that this is probably the first time I've heard someone suggest to push up the Turbines power output xD

I'ts an interesting notion, one that I should think about a bit

**Comment by papaworld** [comment]
For some background, when ErogenousBeef finished with BR 1.7.10, I was considering porting to MC 1.8 (gag), then MC went to 1.9 and you came out with ER for 1.9.4, so I dropped it. But during that time is studied reactors in real world designs:

![2020-11-28_12 29 25](https://user-images.githubusercontent.com/50890375/100523849-2d8dbe80-3179-11eb-8bec-96b64e4542c3.png)

And also how turbine power curves work: (this one is for 1M out)

![12 Turbine 7x7x16 80blade 32coil 3 Combined](https://user-images.githubusercontent.com/50890375/100523859-4007f800-3179-11eb-8b9a-537278036441.jpg)

Then studied the math involved. Then MC went to 1.14.4 and I rethought contacting you for porting ER to 1.14/1.15 but read you were considering doing it on curse forge. So I dropped it for that reason and I really wouldn't have the time to do the mod justice.

The concepts I had for porting were making it as simple of a change over as possible and 2 or 3 tier systems to satisfy both stringent and looser MC players... fair game play balance and also a viable system that would generate compatible(ish) power as Meka.

Any way... I'd be more than happy to bat-around some ideas with ya. ThanX again for bringing ER2. BR/ER have truly one of my favorite and go-to mods.
